Sources/ProjectSpec/SpecFile.swift

```diff
@@ -48,8 +54,11 @@ public struct SpecFile {
         }
     }
 
-    public init(path: Path, cachedSpecFiles: inout [Path: SpecFile], variables: [String: String] = [:]) throws {
-        try self.init(filePath: path, basePath: path.parent(), cachedSpecFiles: &cachedSpecFiles, variables: variables)
+    public init(path: Path, basePath: Path? = nil, cachedSpecFiles: inout [Path: SpecFile], variables: [String: String] = [:]) throws {
+        let basePath = basePath ?? path.parent()
+        let filePath = try path.relativePath(from: basePath)
```

Comment:
Using a relative path ( "-r ." for instance) as a project root results in this line throwing an unmatchedAbsolutePath error. We might want to use basePath?.absolute(), either here or in SpecLoader.

Reply:
That's interesting, I'll try and take a look at that.